### PR TITLE
added comments: trap.c trap.h, removed undifinition function.

### PR DESCRIPTION
**Pull request by kim6515516**
Added comments for doxygen style in trap.[ch].

Removed function : hvmm_status_t trap_hvc_dabort(unsigned int iss, struct arch_regs *regs); in trap.h

### CODE REVIEW

**Comment by kesl** [comment]
Can one of the admins verify this patch?

File: hypervisor/hardware/arm32ve/libhw/trap.c [hunk 1]
```diff
@@ -7,21 +7,44 @@
 #include <vdev.h>
 #define DEBUG
 #include <log/print.h>
+/**\defgroup ARM
+ * <pre> ARM registers.
+ * ARM registers include 13 general purpose registers r0-r12, 1 Stack Pointer,
+ * 1 Link Register (LR), 1 Program Counter (PC).
+ *
```
Comment (Wonseok) [inline hypervisor/hardware/arm32ve/libhw/trap.c:14]: Remove this empty line.

File: hypervisor/hardware/arm32ve/libhw/trap.c [hunk 2]
```diff
@@ -65,6 +93,34 @@ static void _trap_dump_bregs(void)
  * END OF HSR DESCRIPTION FROM ARM DDI0406_C ARCHITECTURE MANUAL
  */
 
+/**@brief Handles the HSR which holds syndrome information for any synchronous
+ * exception taken to Hyp mode. Compared with the reporting of exceptions
+ * taken to PL1 modes.
+ * <pre> Exception class
```
Comment (Wonseok) [inline hypervisor/hardware/arm32ve/libhw/trap.c:99]: Do not need all kind of exception in here.

File: hypervisor/hardware/arm32ve/libhw/trap.c [hunk 3]
```diff
@@ -65,6 +93,34 @@ static void _trap_dump_bregs(void)
...
+ *     Trapped MRRC access to CP14
+ *     Supervisor Call exception routed to Hyp mode
+ *     Hypervisor Call
+ *     Trapped SMC instruction
+ *     Prefetch Abort routed to Hyp mode
+ *     Prefetch Abort taken from Hyp mode
+ *     Data Abort routed to Hyp mode
+ *     Data Abort taken from Hyp mode
+ *</pre>
+ * @param regs ARM registers used by current virtual machine.
```
Comment (Wonseok) [inline hypervisor/hardware/arm32ve/libhw/trap.c:118]: s/ARM registers used by current virtual machine./ARM registers for current virtual machine.

File: hypervisor/hardware/arm32ve/libhw/trap.h [hunk 4]
```diff
@@ -93,7 +93,11 @@
 #define HPFAR_FIPA_PAGE_MASK                0x00000FFF
 #define HPFAR_FIPA_PAGE_SHIFT               12
 
-hvmm_status_t trap_hvc_dabort(unsigned int iss, struct arch_regs *regs);
+/**@brief Handles HSR which holds syndrome information for any synchronous
+ *  exception taken to Hyp mode.
+ * @param regs ARM registers used by current virtual machine.
+ * @return Returns the result of exception using HSR.
+ */
 enum hyp_hvc_result _hyp_hvc_service(struct arch_regs *regs);
```
Comment (Wonseok) [inline hypervisor/hardware/arm32ve/libhw/trap.h:101]: s/Handles HSR/Handles every exceptions taken from a mode other than Hyp mode.

**Comment by Wonseok** [comment]
Please separate this commit into two commits for Added comments for doxygen style in trap.[ch]. and Removed function.

**Pull request closed without merging**